build/azure-pipelines/product-build.yml

```diff
@@ -134,14 +135,30 @@ stages:
     steps:
     - template: darwin/product-build-darwin.yml
 
+- stage: macOSARM64
+  dependsOn:
+  - Compile
+  condition: and(succeeded(), eq(variables['VSCODE_COMPILE_ONLY'], 'false'))
+  pool:
+    vmImage: macOS-latest
+  jobs:
+  - job: macOSARM64
+    condition: and(succeeded(), eq(variables['VSCODE_BUILD_MACOS_ARM64'], 'true'))
```

Comment:
Turning this off for now because internal mac arm builds are not available yet, there is active work to get a release out sometime next week.